### PR TITLE
add shellcheck.exclude option to exclude certain checks

### PR DESCRIPTION
**Pull request by dcepelik**
This allows you to specify excluded checks on a per-`.arclint` basis.
Thanks for the linter and for considering this PR! Let me know if I can do more.

### CODE REVIEW

**Comment by dereckson** [comment]
Thanks for the pull request.

I concur that's a good idea.

**Review by dereckson — APPROVED** [review]
Code is clean and looks good to me.

**Comment by dcepelik** [comment]
Awesome! Anything else I can do?

**Comment by dcepelik** [comment]
@dereckson Can you merge please? :)

**Comment by dereckson** [comment]
Thanks for the ping. I'll merge this this morning.

File: README.md [hunk 1]
```diff
@@ -51,6 +51,11 @@ if you aren't able to install globally as instructed above:
 }
 ```
 
+**OPTIONS**
+
+Please see `arc linters shellcheck` for the list of options supported by this
```
Comment (dereckson) [inline README.md:56]: Works fine:

```
$ arc linters shellcheck
 CONFIGURED  shellcheck (ShellCheck)

    http://www.shellcheck.net/

    ShellCheck is a static analysis and linting tool for sh/bash scripts.

    Configuration Options

    shellcheck.shell (optional string)
      Specify shell dialect (bash, sh, ksh, zsh).

    shellcheck.exclude (optional list<string>)
      Specify excluded checks, e.g.: SC2035.

    bin (optional string | list<string>)
      Specify a string (or list of strings) identifying the binary which
      should be invoked to execute this linter. This overrides the default
      binary. If you provide a list of possible binaries, the first one which
      exists will be used.

    flags (optional list<string>)
      Provide a list of additional flags to pass to the linter on the command
      line.

    version (optional string)
      Specify a version requirement for the binary. The version number may be
      prefixed with <, <=, >, >=, or = to specify the version comparison
      operator (default: =).

    severity (optional map<string|int, string>)
      Provide a map from lint codes to adjusted severity levels: error,
      warning, advice, autofix or disabled.

    severity.rules (optional map<string, string>)
      Provide a map of regular expressions to severity levels. All matching
      codes have their severity adjusted.

    standard (optional string | list<string>)
      The coding standard(s) to apply.
````

**Review by dereckson — APPROVED** [review]
Works fine with last version in master branch (not tested with experimental)

File: lint/linter/ArcanistShellCheckLinter.php [hunk 2]
```diff
@@ -37,6 +38,10 @@ public function getLinterConfigurationOptions() {
           'ksh',
           'zsh'),
       ),
+      'shellcheck.exclude' => array(
```
Comment (dereckson) [inline lint/linter/ArcanistShellCheckLinter.php]: Test:

```name=test-exclude.sh
#!/bin/sh

echo `date`
````
In .arclint:

```
…
        "shell": {
            "type": "shellcheck",
            "include": [
                "(\\.sh$)"
            ],
            "shellcheck.exclude": [
                "SC2046"
            ]
        },
…
```

**Review by dereckson — APPROVED** [review]

**Pull request closed without merging**